### PR TITLE
Text-ui Completion and Suggestion

### PR DESCRIPTION
**Pull request by nkotaa**
# Description
This PR closes the following issues
- #92 
- #47

### CODE REVIEW

File: src/main/java/seedu/binbash/parser/DeleteCommandParser.java
Comment (YHWong20) [inline src/main/java/seedu/binbash/parser/DeleteCommandParser.java:1]: Thanks for implementing the delete functionality using flags!

File: src/main/java/seedu/binbash/ui/Ui.java [hunk 1]
```diff
@@ -32,14 +55,21 @@ public void setUserAsInactive() {
...
         assert isUserActive();
-        return inputReader.nextLine();
+        try {
+            String userInput = inputReader.readLine("binbash> ");
+            UILOGGER.info("received raw user input: " + userInput);
+            return userInput;
+        } catch (EndOfFileException | UserInterruptException e) {
+            UILOGGER.info("received EOF / interrupt exception");
+            return "bye";
+        }
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/ui/Ui.java:65]: Does this mean that `Ctrl + C` will be treated as a `bye` command?

File: src/test/java/seedu/binbash/parser/ParserTest.java [hunk 2]
```diff
@@ -119,17 +109,12 @@ public void testParseCommand_validCommandList_returnsListCommand() throws BinBas
     }
 
     @Test
-    public void testParseCommand_invalidAddCommand_throwsInvalidFormatException() {
-        assertThrows(InvalidFormatException.class, () -> parser.parseCommand("add invalid format"));
-    }
-
-    @Test
-    public void testParseCommand_invalidDeleteCommand_throwsInvalidArgumentException() {
-        assertThrows(InvalidArgumentException.class, () -> parser.parseCommand("delete -1"));
```
Comment (YHWong20) [inline src/test/java/seedu/binbash/parser/ParserTest.java:128]: Should we perhaps retain a unit test here, to ensure that the Parser can recognize invalid/improperly formatted `delete` commands?

File: text-ui-test/EXPECTED.TXT [hunk 3]
```diff
@@ -1,3 +1,5 @@
+Apr 03, 2024 1:00:19 PM seedu.binbash.logger.BinBashLogger consoleLog
+INFO: Data loaded successfully
```
Comment (YHWong20) [inline text-ui-test/EXPECTED.TXT:2]: These two lines shouldn't need to be included in here. Anyway, also note that the date and time is hard-coded into the output.

File: text-ui-test/EXPECTED.TXT [hunk 4]
```diff
@@ -14,10 +16,10 @@ Here are your metrics:
 	Net Profit: 0.00
 
 -------------------------------------------------------------
--------------------------------------------------------------
+binbash> -------------------------------------------------------------
```
Comment (YHWong20) [inline text-ui-test/EXPECTED.TXT:19]: Not sure if the `binbash>` prompt needs to be included as well.

File: text-ui-test/EXPECTED.TXT
Comment (YHWong20) [inline text-ui-test/EXPECTED.TXT:1]: BTW If you're having trouble getting the text UI tests to work then we could always disable this check in Gradle.

**Review by YHWong20 — APPROVED** [review]
Looks good! Thanks for your effort.

File: src/main/java/seedu/binbash/parser/DeleteCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package seedu.binbash.parser;
```
Comment (PureUsagi) [inline src/main/java/seedu/binbash/parser/DeleteCommandParser.java:1]: Thanks for refactoring and creating the DeleteCommandParser!

**Review by PureUsagi — COMMENTED** [review]
Thanks for refactoring the delete command and the UI!

**Pull request merged**